src/mlpack/methods/kde/kde_model.hpp

```diff
@@ -0,0 +1,381 @@
+/**
+ * @file kde_model.hpp
+ * @author Roberto Hueso
+ *
+ * Model for KDE. It abstracts different types of tree, kernels, etc.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_KDE_MODEL_HPP
+#define MLPACK_METHODS_KDE_MODEL_HPP
+
+// Include trees.
+#include <mlpack/core/tree/binary_space_tree.hpp>
+#include <mlpack/core/tree/octree.hpp>
+#include <mlpack/core/tree/cover_tree.hpp>
+#include <mlpack/core/tree/rectangle_tree.hpp>
+
+// Include core.
+#include <mlpack/core.hpp>
+
+// Remaining includes.
+#include <boost/variant.hpp>
+#include "kde.hpp"
+
+namespace mlpack {
+namespace kde {
+
+//! Alias template.
+template<typename KernelType,
+         template<typename TreeMetricType,
+                  typename TreeStatType,
+                  typename TreeMatType> class TreeType>
+using KDEType = KDE<KernelType,
+                    metric::EuclideanDistance,
+                    arma::mat,
+                    TreeType,
+                    TreeType<metric::EuclideanDistance,
+                             kde::KDEStat,
+                             arma::mat>::template DualTreeTraverser,
+                    TreeType<metric::EuclideanDistance,
+                             kde::KDEStat,
+                             arma::mat>::template SingleTreeTraverser>;
+/**
+ * KernerlNormalizer holds a set of methods to normalize estimations applying
```

Comment:
I guess this is `KernelNormalizer`?

Reply:
Exactly :) Fixed typo in 0d5bb4641cc0e1795dc1a17669b4f480275b8854